### PR TITLE
otadump: init at 0.1.2

### PR DESCRIPTION
**Pull request by axelkar**
## Description of changes

It helps to extract partitions from Android OTA files (payload.bin) like payload-dumper-go.

I removed the `static` feature of the rust-lzma crate since the author used it to build universal binaries using musl.

The binary works in `nix-shell --pure -p` too but I'm hoping someone with more experience can tell me if lzma needs to be added to propagatedBuildInputs or something.

<https://github.com/crazystylus/otadump>

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by axelkar** [comment]
@Aleksanaa  You're the maintainer of payload-dumper-go. Can you help review this PR?

File: pkgs/by-name/ot/otadump/package.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
+{ lib, fetchFromGitHub, rustPlatform, pkg-config, protobuf, lzma }:
```
Comment (FliegendeWurst) [inline pkgs/by-name/ot/otadump/package.nix:1]:
```suggestion
{ lib, fetchFromGitHub, rustPlatform, pkg-config, protobuf, xz }:
```
same below

**Review by FliegendeWurst — DISMISSED** [review]

File: pkgs/by-name/ot/otadump/package.nix [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  xz,
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "otadump";
+  version = "0.1.2";
+
+  src = fetchFromGitHub {
+    owner = "crazystylus";
+    repo = pname;
```
Comment (Aleksanaa) [inline pkgs/by-name/ot/otadump/package.nix:16]:
```suggestion
    repo = "otadump";
```

File: pkgs/by-name/ot/otadump/package.nix [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+  doCheck = false; # There are no tests
+
+  cargoHash = "sha256-lTzEgy9mevkmefvTZT9hEBHN5I+kXVqTev5+sy/JoaE=";
+
+  meta = {
+    description = "Command-line tool to extract partitions from Android OTA files";
+    homepage = "https://github.com/crazystylus/otadump";
+    license = lib.licenses.mit;
+    maintainers = [ lib.maintainers.axka ];
```
Comment (Aleksanaa) [inline pkgs/by-name/ot/otadump/package.nix:38]: missing `mainProgram`

Comment (axelkar) [reply pkgs/by-name/ot/otadump/package.nix:38]: Hi, why is mainProgram needed when the pname is the same as the program's name? Is this a new policy in Nixpkgs?

Comment (Aleksanaa) [reply pkgs/by-name/ot/otadump/package.nix:38]: Yes, now all packages must have `mainProgram` set, if there is a main executable. See https://github.com/NixOS/nixpkgs/issues/281276 https://github.com/NixOS/nixpkgs/issues/219567

File: pkgs/by-name/ot/otadump/package.nix [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  };
+
+  patches = [ ./no-static.patch ];
+
+  nativeBuildInputs = [
+    pkg-config
+    protobuf
+  ];
+
+  buildInputs = [ xz.dev ];
```
Comment (Aleksanaa) [inline pkgs/by-name/ot/otadump/package.nix:28]:
```suggestion
  buildInputs = [ xz ];
```

**Review by Aleksanaa — APPROVED** [review]

**Comment by Aleksanaa** [comment]
@ofborg build otadump

**Pull request merged**